test/mock-response-generator.ts

```diff
@@ -1,102 +1,112 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { SearchResponse } from '../src/responses/search/search-response';
+import { SearchResponse } from '../src/responses/search-response';
 import { SearchParams } from '../src/search-params';
-import { Metadata } from '../src/models/metadata';
+import { ItemHit } from '../src/models/hit-types/item-hit';
+import { TextHit } from '../src/models/hit-types/text-hit';
 
 export class MockResponseGenerator {
-  generateMockSearchResponse(params: SearchParams): SearchResponse {
-    const fieldsAsString = params.fields?.join(',');
-
-    const metadata1 = new Metadata({ identifier: 'foo' });
-    const metadata2 = new Metadata({ identifier: 'bar' });
+  generateMockMetadataSearchResponse(params: SearchParams): SearchResponse {
+    const metadata1 = new ItemHit({ identifier: 'foo' });
+    const metadata2 = new ItemHit({ identifier: 'bar' });
 
     return {
       rawResponse: {
         foo: 'bar',
       },
-      responseHeader: {
-        status: 0,
-        QTime: 1459,
-        params: {
-          query: params.query,
-          qin: params.query,
-          fields: fieldsAsString ?? '',
-          wt: 'json',
-          sort: params.sort?.reduce((prev, current, index) => {
-            const isFirst = index === 0;
-            const commaPrefix = isFirst ? '' : ', ';
-            return `${prev}${commaPrefix}${current.field} ${current.direction}`;
-          }, ''),
-          rows: params.rows ? `${params.rows}` : '',
-          start: params.page ?? 0,
+      request: {
+        client_parameters: {
+          client: 'page_production_service_endpoint',
+          user_query: params.query,
+          page: params.page ?? 0,
+          hits_per_page: params.rows ?? 0,
+          fields: params.fields ?? [],
         },
+        finalized_parameters: {
+          client: 'page_production_service_endpoint',
+          user_query: params.query,
+          service_backend: 'metadata',
+          page: params.page ?? 0,
+          page_type: 'search_results',
+          hits_per_page: 50,
+          fields: ['_tile_'].concat(params.fields ?? []),
+          sort: ['_score'],
+          aggregations: [
+            'mediatype',
+            'year',
+            'subject',
+            'collection',
+            'creator',
+            'language',
+          ],
+          aggregations_size: 6,
```

Comment:
QUESTION: Should this number be calculated dynamically from the number of entries in the array assigned to the `aggregations` key?